### PR TITLE
tidy variables, fix loading link

### PR DESCRIPTION
**Pull request by xxyzz**
#110

### CODE REVIEW

**Comment by Anduin2017** [comment]
Seems that when the other side sends a message, the scrolling was not smooth.

When the other side sends a message and the user was not at the bottom, the screen shall not move.

**Review by Anduin2017 — CHANGES_REQUESTED** [review]
Seems that when the other side sends a message, the scrolling was not smooth.

When the other side sends a message and the user was not at the bottom, the screen shall not move.

File: src/app/Controllers/talking.component.ts [hunk 1]
```diff
@@ -172,16 +188,11 @@ export class TalkingComponent implements OnInit, OnDestroy {
...
         }
-        const tempMessage = new Message();
-        tempMessage.content = this.content;
-        tempMessage.sender = AppComponent.me;
-        tempMessage.sender.avatarURL = Values.fileAddress + AppComponent.me.headImgFileKey;
-        tempMessage.senderId = AppComponent.me.id;
-        tempMessage.local = true;
-        this.messages.push(tempMessage);
         this.content = AES.encrypt(this.content, this.conversation.aesKey).toString();
         this.apiService.SendMessage(this.conversation.id, this.content)
```
Comment (Anduin2017) [inline src/app/Controllers/talking.component.ts:192]: When the user scrolls to the bottom of the screen, if he sends a new message, he should immediately scroll the screen to the end to display his new message. When the message is confirmed by the server, nothing should happen. (It seems that the code of the Dev branch is not the effect now)

It seems that when I send a message and the new message was not added to the list.

**Review by Anduin2017 — CHANGES_REQUESTED** [review]
When I opened a new talking page, and then I have received a new message. No matter what I do except quitting the page can not let the red dot disappears.

File: src/app/Controllers/talking.component.ts [hunk 2]
```diff
@@ -112,23 +106,45 @@ export class TalkingComponent implements OnInit, OnDestroy {
...
+                    this.messages = messages;
                 }
-                this.messages = messages;
-                if (getDown && this.onBottom) {
+                if (getDown && this.belowWindowPercent <= 0.2) {
                     setTimeout(() => {
                         this.scrollBottom(true);
                     }, 0);
                 } else if (!getDown) {
                     setTimeout(() => {
```
Comment (Anduin2017) [inline src/app/Controllers/talking.component.ts:123]: I don't agree to scroll when `getDown` was false.

**Review by Anduin2017 — CHANGES_REQUESTED** [review]

File: src/app/Controllers/talking.component.ts [hunk 3]
```diff
@@ -172,16 +188,11 @@ export class TalkingComponent implements OnInit, OnDestroy {
...
-        tempMessage.sender = AppComponent.me;
-        tempMessage.sender.avatarURL = Values.fileAddress + AppComponent.me.headImgFileKey;
-        tempMessage.senderId = AppComponent.me.id;
-        tempMessage.local = true;
-        this.messages.push(tempMessage);
         this.content = AES.encrypt(this.content, this.conversation.aesKey).toString();
         this.apiService.SendMessage(this.conversation.id, this.content)
-            .subscribe(() => { });
+            .subscribe(() => {
+                this.scrollBottom(true);
```
Comment (Anduin2017) [inline src/app/Controllers/talking.component.ts]: Shall not scroll here.

**Review by Anduin2017 — CHANGES_REQUESTED** [review]

File: src/app/Controllers/talking.component.ts (hunk 2 above)
Comment (xxyzz) [reply src/app/Controllers/talking.component.ts:123]: That's for preventing page scroll when loading older messages.

**Review by Anduin2017 — DISMISSED** [review]
The current user enters the talking page, and scroll to the bottom of the page. When the other part sends a message to him, the screen was scrolled from the beginning to the end. Which was not excepted.

The screen shall just scroll down to the bottom and display the latest got message when the current page was already docked to the bottom.

**Review by Anduin2017 — APPROVED** [review]

**Review by Anduin2017 — APPROVED** [review]

**Pull request merged**